app/controllers/concerns/vendor_api/application_data_concerns.rb

```diff
@@ -0,0 +1,45 @@
+module VendorAPI
+  module ApplicationDataConcerns
+  private
+
+    def application_choice_id
+      params.permit(:application_id)[:application_id]
+    end
+
+    def application_choice
+      @application_choice ||= application_choices_visible_to_provider.find(application_choice_id)
+    end
+
+    def render_application
+      render json: application_json
+    end
+
+    def application_json
+      data = { data: ApplicationPresenter.new(version_number, application_choice).as_json }
+      data.merge!(meta) if version_number > '1.0'
+      data.to_json
+    end
+
+    def application_choices_visible_to_provider(includes = nil)
+      options = { providers: [current_provider],
+                  exclude_deferrals: exclude_deferrals }
+      options.merge!(includes: includes) if includes.present?
+
+      GetApplicationChoicesForProviders.call(options)
+    end
+
+    def exclude_deferrals
+      version_number.eql?('1.0')
+    end
+
+    def meta
+      {
+        meta:
```

Comment:
could we add a test for this?

Reply:
Added a ticket so we can tackle this seperately